### PR TITLE
Ce for projections

### PR DESCRIPTION
**Pull request by dtrzesniak**
Here we extend and fix some issues with the projections and forecasting functionality.
 - Filter AY
 - Fix logic for AOC projections
 - Override Valuation Period in getting Values
 - Actuals == Expected for projections

All tests are green

### CODE REVIEW

File: ifrs17/Import/3ImportScope-Actuals.ipynb [hunk 1]
```diff
@@ -173,7 +184,7 @@
                 "\n    [IdentityProperty][NotVisible][Dimension(typeof(EstimateType))]",
                 "\n    string EstimateType => EstimateTypes.A;",
                 "\n    ",
-                "\n    private int?[] accidentYears => GetStorage().GetAccidentYears(Identity.DataNode).ToArray();",
+                "\n    private int?[] accidentYears => GetStorage().GetAccidentYears(Identity.DataNode, Identity.ProjectionPeriod).ToArray();",
                 "\n    ",
                 "\n    [NotVisible]",
                 "\n    ActualBase[] Actuals => GetScope<ValidAmountType>(Identity.DataNode).ActualAmountTypes",
```
Comment (dcolleoni) [inline ifrs17/Import/3ImportScope-Actuals.ipynb:190]: You need to concat the be AmountTypes here

File: ifrs17-template/Test/Data/IfrsVariableBenchmarks/BM_CH_2021_3__C_Projection_DTP.csv [hunk 2]
```diff
@@ -0,0 +1,12 @@
+@@Main,,,,,,,,,,,
+Month,ReportingNode,Scenario,Year,,,,,,,,
+3,CH,,2021,,,,,,,,
+@@C,,,,,,,,,,,
+AccidentYear,AmountType,AocType,DataNode,EconomicBasis,EstimateType,Novelty,Values0,Values1,Values2,Values3,Values4
```
Comment (dcolleoni) [inline ifrs17-template/Test/Data/IfrsVariableBenchmarks/BM_CH_2021_3__C_Projection_DTP.csv:5]: EA and IA steps are missing. 
EA should exists only for the current period.

File: ifrs17-template/Test/Data/IfrsVariableBenchmarks/BM_CH_2021_3__L_Projection_DTP.csv [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+@@L,,,,,,,,,,
+AccidentYear,AmountType,AocType,DataNode,EconomicBasis,EstimateType,Novelty,Values0,Values1,Values2,Values3
+,,AU,DTP1.1,L,L,N,-217.25,-197.5,-177.75,-158
+,,BOP,DTP1.1,L,L,I,260,180.75,0,0
+,,BOP,DTP1.1,L,L,N,60,60,60,60
+,,CL,DTP1.1,L,L,C,-198,-264,-265.25,-266.5
+,,EA,DTP1.1,L,L,C,116,0,0,0
+,,EOP,DTP1.1,L,L,C,180.75,0,0,0
+,,EV,DTP1.1,L,L,N,203.5,185,166.5,148
+,,MC,DTP1.1,L,L,I,-260,-180.75,0,0
```
Comment (dcolleoni) [inline ifrs17-template/Test/Data/IfrsVariableBenchmarks/BM_CH_2021_3__L_Projection_DTP.csv:13]: why MC,I cancels the BOP,I? It looks strange...

File: ifrs17/Import/3ImportScope-Actuals.ipynb (hunk 1 above)
Comment (dcolleoni) [reply ifrs17/Import/3ImportScope-Actuals.ipynb:190]: not necessary for CSM

Comment (dtrzesniak) [reply ifrs17/Import/3ImportScope-Actuals.ipynb:190]: Yes, as discussed, we can do it later.

**Review by dcolleoni — APPROVED** [review]

File: ifrs17-template/Test/Data/IfrsVariableBenchmarks/BM_CH_2021_3__C_Projection_DTP.csv (hunk 2 above)
Comment (dtrzesniak) [reply ifrs17-template/Test/Data/IfrsVariableBenchmarks/BM_CH_2021_3__C_Projection_DTP.csv:5]: Locked in is Zero

**Pull request merged**